pages/advanced/smart-account-modules/smart-account-modules-tutorial.mdx

```diff
@@ -0,0 +1,964 @@
+import { Callout } from 'nextra/components'
+
+# How to build an app with Safe and Safe Module
+
+In this tutorial, you will build an app that can:
+
+- Build a Safe Module
+- Enable an example module on a Safe (the [TokenTransferModule](https://github.com/5afe/safe-module-tutorial-contracts) module)
+- Send a transaction via the TokenTransferModule (Send a transaction that transfers ERC20 token from the Safe account)
+
+The full code for this tutorial is in the [Safe Module tutorial repository](https://github.com/5afe/safe-module-tutorial).
+
+## Prerequisites
+
+**Prerequisite knowledge:** You will need some basic experience with [React](https://react.dev/learn), [Next.js](https://nextjs.org/docs), [Solidity](https://docs.soliditylang.org/en/latest/) and [Hardhat](https://hardhat.org).
+
+Before progressing with the tutorial, please make sure you have the following:
+
+- Downloaded and installed [Node.js](https://nodejs.org/en/download/package-manager) and [pnpm](https://pnpm.io/installation).
+- Metamask installed in your browser and connected to the Sepolia network.
+- Two test accounts in Metamask, the second with some Sepolia Eth for gas.
+
+## Setup contracts repository
+
+This tutorial will provide a guide to create a Safe Module that allows users to withdraw ERC20 tokens from a Safe account, provided that the Safe owners have given their approval in the form of off-chain signatures. The primary use case is to enable beneficiaries to withdraw tokens themselves without requiring the Safe owners to execute transactions directly.
+
+### Use cases
+- Simplifies token withdrawal by enabling beneficiaries to perform withdrawals themselves.
+- Reduces the need for Safe owners to execute each transaction on-chain.
+
+### Limitations
+- Beneficiaries must be externally owned accounts (EOAs) and cannot be smart contracts.
+- The nonce is sequential for simplicity
+- The contract will work only for specific token and Safe address provided during deployment.
+
+### Initialize a new repository
+
+```bash
+git init
+```
+
+```bash
+npm init
+```
+
+### Install dependencies
+
+Add overrides in `package.json` so that there are no peer dependency related issues.
+
+```json
+...
+  "overrides": {
+    "ethers": "^6.13.5"
+  }
+...
+```
+
+```bash
+npm add hardhat @safe-global/safe-contracts @openzeppelin/contracts
+```
+
+### Initialize hardhat project
+
+Select create TypeScript project.
+
+```bash
+npx hardhat init
+```
+
+```bash
+npx hardhat compile
+```
+
+### Update hardhat.config.ts
+
+When compiling Safe contracts with solidity 0.8.x the bytecode size exceeds the limit of 24KB. To overcome this, set `allowUnlimitedContractSize` to `true` in the hardhat config.
```

Comment:
this means after the tutorial the contract will not be deployable to public networks, not sure if this is a good experience

Reply:
In practice, users will only deploy module code and not the Safe contracts.

Reply:
added the comment to use official safe addresses in production.